main.go

```diff
@@ -398,30 +390,33 @@ func usageMessage() {
 /* listAll = true - all versions including beta and rc will be displayed */
 /* listAll = false - only official stable release are displayed */
 func installOption(listAll bool, custBinPath, mirrorURL *string) {
-	tflist, _ := lib.GetTFList(*mirrorURL, listAll) //get list of versions
-	recentVersions, _ := lib.GetRecentVersions()    //get recent versions from RECENT file
-	tflist = append(recentVersions, tflist...)      //append recent versions to the top of the list
-	tflist = lib.RemoveDuplicateVersions(tflist)    //remove duplicate version
+	tflist, _ := lib.GetTFReleases(*mirrorURL, listAll) //get list of versions
+	recentVersions, _ := lib.GetRecentVersions()        //get recent versions from RECENT file
+	var versions []string
+	for _, r := range tflist {
+		versions = append(versions, r.Version)
+	}
+	versions = append(recentVersions, versions...)
+	versions = lib.RemoveDuplicateVersions(versions) //remove duplicate version
 
 	if len(tflist) == 0 {
-		fmt.Println("[ERROR] : List is empty")
-		os.Exit(1)
+		log.Fatalln("[ERROR] : List is empty")
 	}
 	/* prompt user to select version of terraform */
 	prompt := promptui.Select{
 		Label: "Select Terraform version",
-		Items: tflist,
+		Items: versions,
```

Comment:
Is there a way in Go to prompt the version from within `tflist` rather than creating a separate slice?
https://github.com/warrensbox/terraform-switcher/blob/22720e86abd77e94b43e7f6cde8e1fa0c4cea1f8/main.go#L395-L409

Reply:
I haven't used this library before, but it kinda looks like you can do some templating, would probably be overriding it to be `"{{ .Version }}"`.
https://pkg.go.dev/github.com/manifoldco/promptui#Select

But you seem to be using doing some deduping of data coming form `tfList`, so it's not clear that you'd want to use `tfList` without that filtering?

Reply:
I'd rework the removedup function if I could get this part working.

Reply:
Fixed! Thanks for the suggestions